### PR TITLE
Terra keystore

### PR DESCRIPTION
**Pull request by RyanRHall**

### CODE REVIEW

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably #5721) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

File: go.mod [hunk 1]
```diff
@@ -48,6 +50,7 @@ require (
 	github.com/satori/go.uuid v1.2.0
 	github.com/scylladb/go-reflectx v1.0.1
 	github.com/shopspring/decimal v1.3.1
+	github.com/smartcontractkit/chainlink-solana v0.0.0-20211217161610-55e26699acf9
```
Comment (aalu1418) [inline go.mod:53]: I think the CI is failing because of this line. It's the old version and doesn't match the version in `develop`?

File: go.mod [hunk 2]
```diff
@@ -256,3 +262,9 @@ require (
 // If that happens to get released in a 3.X.X version, we can add a constraint to our go.mod
 // for it. If its in 4.X.X, then we need all our transitive deps to upgrade to it.
 replace github.com/dgrijalva/jwt-go => github.com/form3tech-oss/jwt-go v3.2.1+incompatible
+
+// replicating the replace directive on cosmos SDK
+replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
+
+// wut
+replace filippo.io/edwards25519 => filippo.io/edwards25519 v1.0.0-beta.3
```
Comment (aalu1418) [inline go.mod:270]: ?

**Review by aalu1418 — CHANGES_REQUESTED** [review]

**Review by aalu1418 — CHANGES_REQUESTED** [review]
Hmmm i'm not sure the terra transmitter key generates the correct string?

terra addresses start with a `terra...` (there might be something additional to get the proper prefix) let me see if i can find a link

File: core/services/keystore/keys/terrakey/key.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+func (key Key) ID() string {
+	return key.PublicKeyStr()
+}
+
+func (key Key) PublicKey() (pubKey cryptotypes.PubKey) {
+	return key.PubKey()
+}
+
+func (key Key) PublicKeyStr() string {
+	return base58.Encode(key.PubKey().Bytes())
```
Comment (aalu1418) [inline core/services/keystore/keys/terrakey/key.go:66]: the `base58` encoding is just a solana thing not a cosmos/terra thing

i think here's a starting point in the terra-go sdk:
```
// Generate Address from Public Key
addr := msg.AccAddress(privKey.PubKey().Address())
assert.Equal(t, addr.String(), "terra1cevwjzwft3pjuf5nc32d9kyrvh5y7fp9havw7k")
```
https://github.com/terra-money/terra.go

it looks like it points back to some cosmos sdks :)

Comment (RyanRHall) [reply core/services/keystore/keys/terrakey/key.go:66]: ahh right :+1: good catch

Comment (RyanRHall) [reply core/services/keystore/keys/terrakey/key.go]: :+1:

**Review by aalu1418 — APPROVED** [review]
LGTM! 🚀 

(gave it a quick test to create and read keys)

**Pull request merged**